src/Farmer/Common.fs

```diff
@@ -764,32 +764,32 @@ module NetworkSecurity =
         member this.ArmValue = this |> NetworkProtocol.ArmValue
 
     type Port =
-    | Port of uint16
-    | Range of First:uint16 * Last:uint16
-    | AnyPort
+        | Port of uint16
+        | Range of First:uint16 * Last:uint16
+        | AnyPort
+        member this.ArmValue =
+            match this with
+            | Port num -> num |> string
+            | Range (first,last) -> sprintf "%d-%d" first last
+            | AnyPort -> "*"
     module Port =
-        let ArmValue = function
-        | Port num -> num |> string
-        | Range (first,last) -> sprintf "%d-%d" first last
-        | AnyPort -> "*"
-    type Port with
-        member this.ArmValue = this |> Port.ArmValue
+        let ArmValue (port:Port) = port.ArmValue
 
     type Endpoint =
-    | Host of System.Net.IPAddress
-    | Network of IPAddressCidr
-    | Tag of string
-    | AnyEndpoint
+        | Host of Net.IPAddress
+        | Network of IPAddressCidr
+        | Tag of string
+        | AnyEndpoint
+        member this.ArmValue =
+            match this with
+            | Host ip -> string ip
+            | Network cidr -> cidr |> IPAddressCidr.format
+            | Tag tag -> tag
+            | AnyEndpoint -> "*"
     module Endpoint =
-        let ArmValue = function
-        | Host ip -> string ip
-        | Network cidr -> cidr |> IPAddressCidr.format
-        | Tag tag -> tag
-        | AnyEndpoint -> "*"
-
-    type Service =
-    | Service of Name:string * Port
-    | Services of Name:string * Service list
+        let ArmValue (endpoint:Endpoint) = endpoint.ArmValue
+
+    type NetworkService = NetworkService of name:string * Port
```

Comment:
Yep, I agree, but I'm starting to think - at least from a V1 approach - that we should leave the "composability" aspect as a concern for the calling to do i.e. you can easily model that yourself and then when it comes to injecting into Farmer, just flatten the data at that point.

Would that work? We can then consider re-introducing that datatype at a later date.

Reply:
I started backporting it but the code was a bit more complex. I'm not convinced that we should be this opinionated on the type here - at least not yet. We can always add it as an overload and then flatten (or it could be done as an extension method on the class).